### DISCUSSION TITLE
Multiple response binding

### DISCUSSION THREAD

**Discussion by KarlHakanNordgren** [micronaut-projects/micronaut-core#7622] [Q&A] [answered]
Hi! I have client which can receive a number of different DTOs. How can I bind the JSON to the POJO which best matches it? I notice that Custom Argument Binding is described [here](https://docs.micronaut.io/latest/guide/#customArgumentBinding). Does it work similarly with binding responses?

**Comment by yawkat** [comment, marked as answer]
you should do this on the deserialization level. i think jacksons deduction-based polymorphism could do it

**Reply by KarlHakanNordgren** [reply]
Thanks for the above. That was indeed a good idea!
